core/src/fields.rs

```diff
@@ -449,15 +451,13 @@ mod unit {
 
 #[derive(Debug)]
 pub struct MongoFields {
-    dbs: Option<Vec<String>>,
-    current_db: Option<usize>,
-    collections_for_db: Option<Vec<String>>,
-    current_collection: Option<usize>,
-    num_fields_for_collection: Option<usize>,
-    current_schema: Option<Schema>,
-    current_field_for_collection: Option<usize>,
-    db_name_filter: Option<Document>,
+    dbs: Vec<String>,
+    current_db_name: String,
+    collections_for_db: Option<Cursor<CollectionSpecification>>,
```

Comment:
The only reason this is an `Option` is the silly driver won't let us construct an empty `Cursor` D: